apps/snitch_core/lib/core/data/model/option_type.ex

```diff
@@ -42,8 +42,8 @@ defmodule Snitch.Data.Model.OptionType do
   @doc """
   Deletes the OptionType
   """
-  @spec delete(OptionType.t()) :: {:ok, OptionType.t()} | {:error, Ecto.Changeset.t()}
-  def delete(instance) do
-    Repo.delete(instance)
+  @spec delete(integer) :: {:ok, OptionType.t()} | {:error, Ecto.Changeset.t()}
+  def delete(id) when is_integer(id) do
+    QH.delete(OptionType, id, Repo)
```

Comment:
The spec of `QH.delete` must match spec of `OTModel.delete`:
```elixir
# Model.OptionType
  @spec delete(non_neg_integer | struct()) ::
          {:ok, OptionType.t()} | {:error, Ecto.Changeset.t()} | {:error, :not_found}
```
_Psst, that's why I don't like QueryHelper much, instead of being helpful it comes in the way of writing clean code._